pkg/rlptools/limit_index.go

```diff
@@ -129,12 +129,23 @@ func (l *LimitIndex) ToLimits() []limitadorv1alpha1.RateLimit {
 	for gwKey, limitsByDomain := range l.gatewayLimits {
 		for domain, limitList := range limitsByDomain {
 			for idx := range limitList {
+				// Currently, Limitador CRD v0.3.1,
+				// the fields "Variables" and "Conditions" are required
+				variables := limitList[idx].Variables
+				if variables == nil {
+					variables = make([]string, 0)
+				}
+				conditions := limitList[idx].Conditions
+				if conditions == nil {
+					conditions = make([]string, 0)
+				}
+
 				limits = append(limits, limitadorv1alpha1.RateLimit{
 					Namespace:  common.MarshallNamespace(gwKey, domain),
 					MaxValue:   limitList[idx].MaxValue,
 					Seconds:    limitList[idx].Seconds,
-					Variables:  limitList[idx].Variables,
-					Conditions: limitList[idx].Conditions,
+					Variables:  variables,
+					Conditions: conditions,
```

Comment:
Like this:

```suggestion
					Variables: GetDefaultIfNil(limitList[idx].Variables, make([]string,0)),
					Conditions: GetDefaultIfNil(limitList[idx].Conditions, make([]string,0)),
```

Reply:
`Variables` is not a pointer, it's a slice.

TRying to compile it complains:
```
pkg/rlptools/limit_index.go:134:41: type []string of limitList[idx].Variables does not match *T (cannot infer T)
```
I can try defining a new method for slices

Reply:
done

Reply:
Will it fail without the `GetEmptySliceIfNil` at all?

If `Variables` and `Conditions` are slices rather than pointers, then both `len` and `range` should work even when the values were omitted in the CR, no?

Reply:
sorry, I do not follow. You mean using the `GetDefaultIfNil` func?

Reply:
I mean if we do

```go
Variables: limitList[idx].Variables,
Conditions: limitList[idx].Conditions,
```

Will it fail for the minimum RLP that does not include neither `variables` nor `conditions`? I'm guessing it will not, but I may be wrong.

Reply:
Yeah, it cannot be `nil`. Hence that change in the code. Limitador's CRD does not allow it to be empty, it is required to be empty at least

Reply:
Maybe you could've used

```suggestion
					Variables:  common.GetDefaultIfNil(&limitList[idx].Variables, make([]string, 0)),
					Conditions: common.GetDefaultIfNil(&limitList[idx].Conditions, make([]string, 0)),
```

That should compile, or I'm missing something?